pkg/yum/yum.go

```diff
@@ -53,27 +54,51 @@ func ExtractPackageData(client http.Client, url string) ([]Package, error) {
 	if resp.StatusCode != http.StatusOK {
 		return []Package{}, fmt.Errorf("Status error: %v", resp.StatusCode)
 	}
-	primaryURL, err := GetPrimaryURLFromRepomdXML(resp.Body, url)
+	primaryURL, _, err := ParseRepomdXML(resp.Body, url)
 
 	resp.Body.Close()
 
 	if err != nil {
 		return []Package{}, err
 	}
 
-	return GetPackagesArrayWithPrimaryURL(client, primaryURL)
+	packages, err := GetPackagesArrayWithPrimaryURL(client, primaryURL)
+
+	return packages, err
+}
+
+func GetRevision(client http.Client, url string) (string, error) {
+	resp, err := client.Get(fmt.Sprintf("%s/repodata/repomd.xml", url))
+	if err != nil {
+		return "", fmt.Errorf("GET error: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("Status error: %v", resp.StatusCode)
+	}
+
+	_, revision, err := ParseRepomdXML(resp.Body, url)
+
+	resp.Body.Close()
+
+	if err != nil {
+		return "", err
+	}
+
+	return revision, err
 }
 
-func GetPrimaryURLFromRepomdXML(body io.ReadCloser, url string) (string, error) {
+// ParseRepomdXML returns URL of primary data and revision number
+func ParseRepomdXML(body io.ReadCloser, url string) (string, string, error) {
```

Comment:
This might break things, but would it make sense to return the 'repomd' struct directly?

Reply:
I changed that then ended up changing a few more things to make it all fit together a little nicer